server/api/repo.go

```diff
@@ -575,3 +520,91 @@ func GetAllRepos(c *gin.Context) {
 
 	c.JSON(http.StatusOK, repos)
 }
+
+// RepairAllRepos
+//
+//	@Summary	Repair all repositories on the server. Requires admin rights.
+//	@Router		/repos/repair [post]
+//	@Produce	plain
+//	@Success	204
+//	@Tags		Repositories
+//	@Param		Authorization	header	string	true	"Insert your personal access token"	default(Bearer <personal access token>)
+func RepairAllRepos(c *gin.Context) {
+	_store := store.FromContext(c)
+
+	repos, err := _store.RepoListAll(true, &model.ListOptions{All: true})
+	if err != nil {
+		c.String(http.StatusInternalServerError, "Error fetching repository list. %s", err)
+		return
+	}
+
+	for _, r := range repos {
+		repairRepo(c, r, false)
+		if c.Writer.Written() {
+			return
+		}
+	}
+
+	c.Status(http.StatusNoContent)
+}
+
+func repairRepo(c *gin.Context, repo *model.Repo, withPerms bool) {
+	forge := server.Config.Services.Forge
+	_store := store.FromContext(c)
+	user := session.User(c)
```

Comment:
Uh the woodpecker admin could be a normal user on the forge ... this case will break things.

For single repair we should alter the user as we do now (witch is used to sync repo stuff) but for itterating over all of them we should use the orig one ...